stablehlo/dialect/StablehloOps.td

```diff
@@ -2788,12 +2719,6 @@ def StableHLO_DynamicGatherOp: StableHLO_Op<"dynamic_gather",
     DefaultValuedAttr<BoolAttr, "false">:$indices_are_sorted
   );
   let results = (outs HLO_Tensor);
-
-  let extraClassDeclaration = [{
-    static bool isCompatibleReturnTypes(TypeRange l, TypeRange r) {
-      return succeeded(mlir::verifyCompatibleShapes(l, r));
-    }
-  }];
 }
 
 def StableHLO_DynamicConvOp : StableHLO_Op<"dynamic_conv", [NoSideEffect]> {
```

Comment:
By the way, I've just realized that [`HLO_PairwiseSameOperandAndResultType`](https://github.com/openxla/stablehlo/blob/388b134277b19b38293354cf7ae4ac0ad4b37447/stablehlo/dialect/Base.td#L179-L180) is still problematic. It needs to be changed to `HLO_PairwiseCompatibleOperandAndResultType` to implement the intent of this PR.

Reply:
We can land this PR as is, once the previous comment about deleting comments is addressed, and work on this comment in https://github.com/openxla/stablehlo/pull/207.

Reply:
Oh sure. Still, let me change `[HLO_PairwiseSameOperandAndResultType](https://github.com/openxla/stablehlo/blob/388b134277b19b38293354cf7ae4ac0ad4b37447/stablehlo/dialect/Base.td#L179-L180)` to `HLO_PairwiseCompatibleOperandAndResultType` to make this PR consistent.